tools/crm_resource_runtime.c

```diff
@@ -1750,3 +1752,87 @@ cli_resource_move(const char *rsc_id, const char *host_name, cib_t * cib, pe_wor
 
     return rc;
 }
+
+static void
+cli_resource_why_without_rsc_and_host(cib_t *cib_conn,GListPtr resources)
+{
+    GListPtr lpc = NULL;
+    GListPtr hosts = NULL;
+
+    for (lpc = resources; lpc != NULL; lpc = lpc->next) {
+        resource_t *rsc = (resource_t *) lpc->data;
+        rsc->fns->location(rsc, &hosts, TRUE);
+	if ( hosts == NULL ) {
+	    printf("Resource %s is not running",rsc->id);
+            cli_resource_check(cib_conn, rsc);
+	} else {
+	    printf("Resource %s is running",rsc->id);
+	}
+     }
+
+     g_list_free(hosts);
+     hosts = NULL;
+}
+
+static void
+cli_resource_why_with_rsc_and_host(cib_t *cib_conn,GListPtr resources,const char* rsc_id,const char* host_uname)
+{
+    resource_t *rsc = NULL;
+
+    rsc = pe_find_resource(resources, rsc_id);
+    if(!(resource_is_running_on(rsc,host_uname))) {
+        cli_resource_check(cib_conn, rsc);
+    } else {
+        printf("Resource %s is not running on host %s",rsc->id,host_uname);
+    }
+}
+
+static void 
+cli_resource_why_without_rsc_with_host(cib_t *cib_conn,GListPtr resources,const char* host_uname)
+{
+    GListPtr allResources = get_all_resources(host_uname,resources);
+    GListPtr activeResources = get_active_resources(host_uname,allResources);
+    GListPtr unactiveResources = subtract_lists(allResources,activeResources);
+    GListPtr lpc = NULL;
+
+    for(lpc = unactiveResources; lpc != NULL; lpc = lpc->next) {
+        resource_t *rsc = (resource_t *) lpc->data;
+        cli_resource_check(cib_conn,rsc);
+     }
```

Comment:
After thinking more about this, I think we should do two loops, through active and inactive resources. The active loop can print the "is running on" message, and the inactive loop can print that the resource "is assigned to NODE-NAME but not running". Both loops should call cli_resource_check().

That way, if someone runs "crm_resource --why -N node1", they get a full list of all resources assigned to the node, and whether each one is running or not running (and possibly why or why not).